documentation/src/test/java/example/TestReporterDemo.java

```diff
@@ -41,5 +46,18 @@ void reportMultipleKeyValuePairs(TestReporter testReporter) {
 		testReporter.publishEntry(values);
 	}
 
+	@Test
+	void reportFiles(TestReporter testReporter, @TempDir Path tempDir) throws Exception {
+
+		testReporter.publishFile("test1.txt", file -> Files.write(file, singletonList("Test 1")));
+
+		Path existingFile = tempDir.resolve("test2.txt");
+		testReporter.publishFile(Files.write(existingFile, singletonList("Test 2")));
+
+		testReporter.publishFile("test3", dir -> {
+			Path nestedFile = Files.createDirectory(dir).resolve("nested.txt");
+			Files.write(nestedFile, singletonList("Nested content"));
+		});
+	}
```

Comment:
This demonstrates what the programming model for test written with JUnit Jupiter would look like.